src/tools/rust-analyzer/crates/proc-macro-api/src/version.rs

```diff
@@ -144,13 +150,13 @@ pub fn read_version(dylib_path: &AbsPath) -> io::Result<String> {
 
     // the bytes before version string bytes, so this basically is:
     // 8 bytes for [b'r',b'u',b's',b't',0,0,0,5]
-    // 4 bytes for [crate root bytes]
+    // 8 bytes for [crate root bytes]
     // 1 byte for length of version string
-    // so 13 bytes in total, and we should check the 13th byte
+    // so 17 bytes in total, and we should check the 17th byte
     // to know the length
-    let mut bytes_before_version = [0u8; 13];
+    let mut bytes_before_version = [0u8; 17];
```

Comment:
This breaks support for previous versions, right?

The server will generally match the compiler, but this might still happen with already-built projects.

Reply:
I think I patched this up by passing more data out of the version match above?

Reply:
It probably doesn't matter too much, but I _think_ you could leave that as an array, and only read into a prefix slice of it?

Reply:
That feels like too much brainpower for this code to me.